webnn/clamp.https.any.js

```diff
@@ -0,0 +1,214 @@
+    -8.275078773498535,  -1.355708122253418, 7,                  -5.530012607574463,
+  ],
+};
+// expected data by clamping input data with specified maxValu=0
+const expectedMaxZero = {
+  float32: [
+    -3.4356448650360107, -6.530945777893066, -8.175760269165039, 0,
+    -4.48015022277832,   -8.591504096984863, 0,                  -6.618697643280029,
+    0,                   0,                  -8.79992389678955,  -3.3445966243743896,
+    0,                   0,                  0,                  0,
+    -7.302720069885254,  0,                  0,                  0,
+    -8.275078773498535,  -1.355708122253418, 0,                  -5.530012607574463,
+  ],
+};
+
+// test = {
+//   purpose: [
+//     [inputShape, expectedData, options],
+//   ],
+// };
+const tests = {
+  'default options': [
+    // 1D
+    [[24], expectedDefault],
+    // 2D
+    [[4, 6], expectedDefault],
+    // 3D
+    [[2, 3, 4], expectedDefault],
+    // 4D
+    [[2, 3, 2, 2], expectedDefault],
+    // 5D
+    [[2, 3, 2, 1, 2], expectedDefault],
+  ],
+  'only specified minValue option': [
+    // 1D
+    [[24], expectedMin, {minValue: -5}],
+    [[24], expectedMinZero, {minValue: 0}],
+    // 2D
+    [[4, 6], expectedMin, {minValue: -5}],
+    [[4, 6], expectedMinZero, {minValue: 0}],
+    // 3D
+    [[2, 3, 4], expectedMin, {minValue: -5}],
+    [[2, 3, 4], expectedMinZero, {minValue: 0}],
+    // 4D
+    [[2, 3, 2, 2], expectedMin, {minValue: -5}],
+    [[2, 3, 2, 2], expectedMinZero, {minValue: 0}],
+    // 5D
+    [[2, 3, 2, 1, 2], expectedMin, {minValue: -5}],
+    [[2, 3, 2, 1, 2], expectedMinZero, {minValue: 0}],
+  ],
+  'only specified maxValue option': [
+    // 1D
+    [[24], expectedMax, {maxValue: 7}],
+    [[24], expectedMaxZero, {maxValue: 0}],
+    // 2D
+    [[4, 6], expectedMax, {maxValue: 7}],
+    [[4, 6], expectedMaxZero, {maxValue: 0}],
+    // 3D
+    [[2, 3, 4], expectedMax, {maxValue: 7}],
+    [[2, 3, 4], expectedMaxZero, {maxValue: 0}],
+    // 4D
+    [[2, 3, 2, 2], expectedMax, {maxValue: 7}],
+    [[2, 3, 2, 2], expectedMaxZero, {maxValue: 0}],
+    // 5D
+    [[2, 3, 2, 1, 2], expectedMax, {maxValue: 7}],
+    [[2, 3, 2, 1, 2], expectedMaxZero, {maxValue: 0}],
+  ],
+  'both specified minValue and maxValue options': [
+    // 1D
+    [[24], expectedMinMax, {minValue: -5, maxValue: 7}],
+    [[24], expectedRelu6, {minValue: 0, maxValue: 6}],
+    // 2D
+    [[4, 6], expectedMinMax, {minValue: -5, maxValue: 7}],
+    [[4, 6], expectedRelu6, {minValue: 0, maxValue: 6}],
+    // 3D
+    [[2, 3, 4], expectedMinMax, {minValue: -5, maxValue: 7}],
+    [[2, 3, 4], expectedRelu6, {minValue: 0, maxValue: 6}],
+    // 4D
+    [[2, 3, 2, 2], expectedMinMax, {minValue: -5, maxValue: 7}],
+    [[2, 3, 2, 2], expectedRelu6, {minValue: 0, maxValue: 6}],
+    // 5D
+    [[2, 3, 2, 1, 2], expectedMinMax, {minValue: -5, maxValue: 7}],
+    [[2, 3, 2, 1, 2], expectedRelu6, {minValue: 0, maxValue: 6}],
+  ],
```

Comment:
Should we have at least one case where minValue > 0 and maxValue < 0?

Reply:
Do you mean adding exception tests? It's expected that an exception error should be throwed for this `minValue > 0 and maxValue < 0` case.

Reply:
e.g.

`   [[2, 3, 2, 1, 2], expectedMinMaxBothPositive, {minValue: 5, maxValue: 10}],`
`   [[2, 3, 2, 1, 2], expectedMinMaxBothNegative, {minValue: -10, maxValue: -5}],``

Reply:
I updated clamp tests by adding such tests, please take a look again, thanks.